MoesifWorker.js

```diff
@@ -471,25 +528,49 @@ async function logRequest(event) {
   } else {
     moesifLog('No response logged');
   }
-  const after = new Date();
-  const txId = request.headers.get(TRANSACTION_ID_HEADER) || uuid4();
-
-  event.waitUntil(
-    tryTrackRequest(
-      event,
-      request,
-      (response ? response : new EmptyResponse()),
-      before,
-      after,
-      txId
-    )
+
+  let userId = null;
+  userId = runHook(
+    () => identifyUser(request, response),
+    'identifyUser',
+    undefined
   );
 
-  if (!disableTransactionId && response && response.body) {
-    const responseClone = new Response(response.body, response);
-    responseClone.headers.set(TRANSACTION_ID_HEADER, txId);
-    return responseClone;
-  } else {
+  let companyId = null;
+  companyId = runHook(
+    () => identifyCompany(request, response),
+    'identifyCompany',
+    undefined
+  );
+
+  samplingPercentage = getSamplingPercentage(userId, companyId);
+  if (randomNumber > samplingPercentage) {
+    moesifLog('Skip sending event to Moesif due to sampling percentage');
     return response;
+  } else {
+    const after = new Date();
+    const txId = request.headers.get(TRANSACTION_ID_HEADER) || uuid4();
+  
+    event.waitUntil(
+      tryTrackRequest(
```

Comment:
can you add `weight` to the event, its needed for various purposes in backend/frontend

Reply:
added weight to the event model